modules/3d/src/octree.cpp

```diff
@@ -0,0 +1,548 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+#include "precomp.hpp"
+#include "octree.hpp"
+
+#define OCTREE_CHILD_NUM 8
+
+namespace cv{
+
+// Locate the OctreeNode corresponding to the input point from the given OctreeNode.
+Ptr<OctreeNode> index(const Point3f& point, Ptr<OctreeNode>& node);
+
+bool _isPointInBound(const Point3f& _point, const Point3f& _origin, double _size);
+void insertPointRecurse( Ptr<OctreeNode>& node,  const Point3f& point, int maxDepth);
+bool deletePointRecurse( Ptr<OctreeNode>& node);
+
+// For Nearest neighbor search.
+float SquaredDistance(const Point3f& query, const Point3f& origin);
+bool overlap(const OctreeNode& node, const Point3f& query, float squareRadius);
+template<typename T> struct PQueueElem; // Priority queue
+void radiusNNSearchRecurse(const Ptr<OctreeNode>& node, const Point3f& query, float squareRadius,
+                           std::vector<PQueueElem<Point3f>>& candidatePoint);
+void KNNSearchRecurse(const Ptr<OctreeNode>& node, const Point3f& query, const int K,
+                      float& smallestDist, std::vector<PQueueElem<Point3f>>& candidatePoint);
```

Comment:
These declarations should be local - use `static` for that.
Also some declaration can be removed (used after declaration) - add `static` to definition only.

Reply:
Thank you for your carefully reviewing. I have added ``static`` prefix in these functions.